compiler/optimizer/VPHandlers.cpp

```diff
@@ -7004,6 +7004,62 @@ TR::Node *constrainLrem(OMR::ValuePropagation *vp, TR::Node *node)
    return node;
    }
 
+TR::Node *constrainLurem(OMR::ValuePropagation *vp, TR::Node *node)
+   {
+   if (findConstant(vp, node))
+      {
+      return node;
+      }
+   constrainChildren(vp, node);
+
+   bool lhsGlobal, rhsGlobal;
+   TR::VPConstraint *lhs = vp->getConstraint(node->getFirstChild(), lhsGlobal);
+   TR::VPConstraint *rhs = vp->getConstraint(node->getSecondChild(), rhsGlobal);
+   TR::VPConstraint *constraint = NULL;
+   lhsGlobal &= rhsGlobal;
+   if (lhs && lhs->asLongConst() &&
+       rhs && rhs->asLongConst())
+      {
+      //
+      // REMOVED AS IT WOULD LEAD TO AN ASSERT POST FORK
+      //
+      TR_ASSERT(0, "constrainLurem not supported");
```

Comment:
This comment and assert needs to be made more relevant in the open-source world.  It's really hiding a limitation of the implementation, namely that we can't fold constant unsigned dividend and divisors for long remainder at compile-time (I think that just needs to be implemented).  I would like to either see the complete implementation here, or have this code fall-back on the most general constraint with an issue created to add unsigned long remainder support to the ArithEnv class.